src/react-render-perf/lesson-04/solution/fib.tsx

```diff
@@ -0,0 +1,13 @@
+import * as React from "react";
+
+import {fib} from "../../shared/fib";
+
+type Props = {n: number};
+
+export default React.memo(function Fib({n}: Props) {
+    return (
+        <p>
+            fib({n}) = {fib(n)}
+        </p>
+    );
+});
```

Comment:
I was poking around some more! We don't use the `arePropsEqual` arg here because none of the props are objects or functions, right?

Also, since the Fib component is memoized now, I was thinking the component wouldn't re-render if the props (the `n` value) didn't change. The visual rendering feature shows that the Fib component gets re-rendered whenever the button is pressed, but the profiler says the Fib component wasn't re-rendered. So the visual rendering feature might not be accurate at times! 

https://github.com/Khan/frontend-infra-workshops/assets/14334617/21fc4642-7d57-4c7f-9e7e-7a8d0318049e

Reply:
Interesting.  I'll have a closer look tomorrow.  Thanks for the video recording.

Reply:
I'm seeing the same behaviour.  I'll make a note about it in the preliminaries.